src/wp-includes/class-wp-theme.php

```diff
@@ -1967,14 +1967,22 @@ public function get_block_patterns() {
 	 * Gets block pattern cache.
 	 *
 	 * @since 6.4.0
+	 * @since 6.6.0 Adds support for transient caching.
 	 *
 	 * @return array|false Returns an array of patterns if cache is found, otherwise false.
 	 */
 	private function get_pattern_cache() {
 		if ( ! $this->exists() ) {
 			return false;
 		}
-		$pattern_data = wp_cache_get( 'wp_theme_patterns_' . $this->stylesheet, 'theme_files' );
+
+		$cache_key = 'wp_theme_patterns_' . md5( $this->stylesheet );
```

Comment:
Since we're changing the cache keys in this PR already, it would make more sense to align it with how the class otherwise handles this:
* Use `$this->cache_hash` instead of `md5( $this->stylesheet )`. This is not only more efficient, but also more reliable as it prevents edge-case bugs where the same theme may be loaded from a different themes folder (since `$this->cache_hash` also takes the "theme root" into account).
* Use `wp_theme_patterns-` (ending in a hyphen instead of underscore). This is just a nit-pick, but brings consistency with how cache keys are connected to the hash elsewhere in the class.

Reply:
Done.